Source/automap.cpp

```diff
@@ -291,114 +249,119 @@ void SearchAutomapItem(const CelOutputBuffer &out)
-		DrawLineTo(out, x + AmLine16, y - AmLine8, x + AmLine8 + AmLine4, y, playerColor);
+		point = { base.x + AmLine16, base.y - AmLine8 };
+		left = { base.x + AmLine8, base.y - AmLine8 };
+		right = { base.x + AmLine8 + AmLine4, base.y };
 		break;
 	case DIR_E:
-		DrawLineTo(out, x, y, x + AmLine16, y, playerColor);
-		DrawLineTo(out, x + AmLine16, y, x + AmLine8, y - AmLine4, playerColor);
-		DrawLineTo(out, x + AmLine16, y, x + AmLine8, y + AmLine4, playerColor);
+		point = { base.x + AmLine16, base.y };
+		left = { base.x + AmLine8, base.y - AmLine4 };
+		right = { base.x + AmLine8, base.y + AmLine4 };
 		break;
 	case DIR_SE:
-		DrawLineTo(out, x, y, x + AmLine16, y + AmLine8, playerColor);
-		DrawLineTo(out, x + AmLine16, y + AmLine8, x + AmLine8 + AmLine4, y, playerColor);
-		DrawLineTo(out, x + AmLine16, y + AmLine8, x + AmLine8, y + AmLine8, playerColor);
+		point = { base.x + AmLine16, base.y + AmLine8 };
+		left = { base.x + AmLine8 + AmLine4, base.y };
+		right = { base.x + AmLine8, base.y + AmLine8 };
 		break;
 	case DIR_S:
 	case DIR_OMNI:
-		DrawLineTo(out, x, y, x, y + AmLine16, playerColor);
-		DrawLineTo(out, x, y + AmLine16, x + AmLine4, y + AmLine8, playerColor);
-		DrawLineTo(out, x, y + AmLine16, x - AmLine4, y + AmLine8, playerColor);
+		point = { base.x, base.y + AmLine16 };
+		left = { base.x + AmLine4, base.y + AmLine8 };
+		right = { base.x - AmLine4, base.y + AmLine8 };
 		break;
 	case DIR_SW:
-		DrawLineTo(out, x, y, x - AmLine16, y + AmLine8, playerColor);
-		DrawLineTo(out, x - AmLine16, y + AmLine8, x - AmLine4 - AmLine8, y, playerColor);
-		DrawLineTo(out, x - AmLine16, y + AmLine8, x - AmLine8, y + AmLine8, playerColor);
+		point = { base.x - AmLine16, base.y + AmLine8 };
+		left = { base.x - AmLine4 - AmLine8, base.y };
+		right = { base.x - AmLine8, base.y + AmLine8 };
 		break;
 	case DIR_W:
-		DrawLineTo(out, x, y, x - AmLine16, y, playerColor);
-		DrawLineTo(out, x - AmLine16, y, x - AmLine8, y - AmLine4, playerColor);
-		DrawLineTo(out, x - AmLine16, y, x - AmLine8, y + AmLine4, playerColor);
+		point = { base.x - AmLine16, base.y };
+		left = { base.x - AmLine8, base.y - AmLine4 };
+		right = { base.x - AmLine8, base.y + AmLine4 };
 		break;
 	case DIR_NW:
-		DrawLineTo(out, x, y, x - AmLine16, y - AmLine8, playerColor);
-		DrawLineTo(out, x - AmLine16, y - AmLine8, x - AmLine8, y - AmLine8, playerColor);
-		DrawLineTo(out, x - AmLine16, y - AmLine8, x - AmLine4 - AmLine8, y, playerColor);
+		point = { base.x - AmLine16, base.y - AmLine8 };
+		left = { base.x - AmLine8, base.y - AmLine8 };
+		right = { base.x - AmLine4 - AmLine8, base.y };
 		break;
 	}
+
+	DrawLineTo(out, base, point, playerColor);
+	DrawLineTo(out, point, left, playerColor);
+	DrawLineTo(out, point, right, playerColor);
 }
 
 /**
  * @brief Returns the automap shape at the given coordinate.
  */
-uint16_t GetAutomapType(int x, int y, bool view)
+uint16_t GetAutomapType(Point Macro, bool view)

```

Comment:
Why `Macro` and not `macro`? Also, what's a `macro`?

Reply:
macro tiles are tiles constructed out of 2x2 tile and is the scale that the automap operates on.

Reply:
Renamed it to map, that should make more sens maybe.

Reply:
`Point map`? So it's not a position? I'm a bit confused.

Reply:
Ah so it's the coordinates of a macro?

Reply:
> `Point map`? So it's not a position? I'm a bit confused.

Being an x,y struct makes it automatically a position in my head. During usage (`map.x, map.y`) it makes sense imo?

Reply:
Hm. A `Point` is not always a position. I don't find it obvious, but it's subjective. I tend to prefer longer and more explicit variables, like `mapPosition` or `mapPos`.

Reply:
The automap context helps a bit. But yeah defiantly subjective.

Reply:
But should objectivly be a step up from just `x, y` I think :)